### PR TITLE
#159569285 Create default loading message while asset categories are being fetched on dashboard

### PR DESCRIPTION
**Pull request by Billmike**
**What does this PR do?**
Creates a graphic spinner while asset categories are being loaded on the dashboard

**Description of Task to be completed?**
- create spinner component
- write actions type to listen for requests to fetch asset categories
- write reducer case for request action dispatched
- write tests for spinner logic

**How should this be manually tested?**
After cloning, run the app by running `yarn start:dev`
- While the product is trying to make an API call, you should see a simple spinner loading.

**Any background context you want to provide?**
Users shouldn't experience a lag in the application due to the time it takes to fetch data, as such, a loader needed to be implemented.

**What are the relevant pivotal tracker stories?**
[#159569285](https://www.pivotaltracker.com/story/show/159569285)

**Screenshots (if appropriate)**
screen shot 2018-08-06 at 8 50 27 pm

**Questions:**
N/A

### CODE REVIEW

File: src/components/Spinner/Spinner.test.tsx [hunk 1]
```diff
@@ -0,0 +1,17 @@
+// import react libraries
+import * as React from 'react';
+
+// import third-party libraries
+import { shallow } from 'enzyme';
+
+// import components
+import Spinner from './';
+
+describe('<Spinner />', () => {
```
Comment (PaulKariukiRimiru) [inline src/components/Spinner/Spinner.test.tsx]: I think that the tests should read as a sentence. How about renaming this to `The Spinner Component` instead.

Comment (Billmike) [reply src/components/Spinner/Spinner.test.tsx]: It does read as a sentence. The `<Spinner />` connotes `The Spinner Component`, as can be seen from the enzyme documentation [here](https://airbnb.io/enzyme/docs/api/shallow.html) 😁

File: src/components/Spinner/Spinner.test.tsx [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  it('should be mounted properly', () => {
+    const wrapper = shallow(<Spinner />);
+
+    expect(wrapper).toMatchSnapshot();
+    expect(wrapper.find('div').length).toEqual(1);
```
Comment (andela-kakpobome) [inline src/components/Spinner/Spinner.test.tsx]: Shouldn't you be testing for the `spinner` class, to make this more specific

File: src/components/Spinner/index.tsx [hunk 3]
```diff
@@ -0,0 +1,14 @@
+// import react library
+import * as React from 'react';
+
+// import styles
+import './Spinner.scss';
+
+const Spinner = () => {
+  return (
+    <div className="spinner">
+    </div>
```
Comment (andela-kakpobome) [inline src/components/Spinner/index.tsx]: This can be on one line

File: src/components/Spinner/Spinner.scss [hunk 4]
```diff
@@ -0,0 +1,24 @@
+@import '../../assets/scss/variables';
+@import 'sass-rem'; 
+
+.spinner::before {
+  content: '';
+  position: absolute;
+  top: 50%;
```
Comment (andela-kakpobome) [inline src/components/Spinner/Spinner.scss]: Let this value be `35%`

**Comment by Billmike** [comment]
@andela-kakpobome All requested changes have been effected.

**Pull request merged**